de.metas.business.rest-api-impl/src/test/java/de/metas/rest_api/invoicecandidates/impl/InvoiceCandidatesRestControllerImplTest.java

```diff
@@ -13,48 +13,53 @@
 import org.junit.Test;
 
 import de.metas.invoicecandidate.InvoiceCandidateId;
+import de.metas.invoicecandidate.api.IInvoiceCandBL;
 import de.metas.invoicecandidate.model.I_C_Invoice_Candidate;
 import de.metas.process.PInstanceId;
 import de.metas.rest_api.invoicecandidates.request.JsonInvoiceCandidate;
+import de.metas.util.Services;
+import de.metas.util.rest.ExternalHeaderAndLineId;
 import de.metas.util.rest.ExternalId;
 
 public class InvoiceCandidatesRestControllerImplTest {
 
 	private static final String EXTERNAL_LINE_ID1 = "Test1";
 	private static final String EXTERNAL_HEADER_ID1 = "1001";
-	
+
 	private static final String EXTERNAL_LINE_ID2 = "Test2";
 	private static final String EXTERNAL_HEADER_ID2 = "1002";
-	
+
 	private static final String EXTERNAL_LINE_ID3 = "Test3";
 	private static final String EXTERNAL_HEADER_ID3 = "1003";
-	
+
 	private static final int P_INSTANCE_ID = 1002265;
-	private InvoiceCandidatesRestControllerImpl invoiceCandidatesRestController;
-	private List<JsonInvoiceCandidate> jsonInvoiceCandidates;
+	private IInvoiceCandBL invoiceCandBL;
+	private InvoiceJsonConverterService jsonConverter;
 
 	@Before
 	public void init() {
```

Comment:
as talked
>  i wanted to extend the unit test a bit; imo where need to be >2 ICs and the method needs to return some (or one) of them. as was as i see, the tests don't do that yet.

also, since the selection is now done in IInvoiceCandBL, i think there should rather be one test for InvoiceJsonConverterService (here) and a nother one for the neew method implementation in `InvoiceCandBL`.

But i think this also isn't a show-stopper..i think we'll do it this week when we continue with invoice candidate REST features

Reply:
Currently InvoiceJsonConverterService is tested in this test and imo another separate test would look like this: 
![image](https://user-images.githubusercontent.com/50820854/67665204-ed567e00-f971-11e9-95dc-a4af0aa43d0b.png)
and it would have a lot of duplicate code. But as you wish, I can create one.